### PR TITLE
break out many parts of common.bzl into separate files and update them to the latest

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: dart/build_rules/common/context.bzl [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  # Merge Dart context by package.
+  ctx_map = {}
+  for dc in dart_ctxs:
+    if dc.package in ctx_map:
+      dc = _merge_dart_context(ctx_map[dc.package], dc)
+    ctx_map[dc.package] = dc
+  return ctx_map
+
+def make_dart_context(ctx,
```
Comment (natebosch) [inline dart/build_rules/common/context.bzl]: while moving this should we update to 4 space indent args and spaces around '='? It'll save us from a second ugly diff later

File: dart/build_rules/common/util.bzl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Common utilities that don't fit in one of the other categories."""
+
+def filter_files(filetypes, files):
```
Comment (natebosch) [inline dart/build_rules/common/util.bzl]: this one fits with path.bzl IMO

File: dart/build_rules/common/util.bzl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  filtered_files = []
+  for file_to_filter in files:
+    for filetype in filetypes:
+      if str(file_to_filter).endswith(filetype):
+        filtered_files.append(file_to_filter)
+        break
+
+  return filtered_files
+
+def label_to_dart_package_name(label):
```
Comment (natebosch) [inline dart/build_rules/common/util.bzl]: this one could fit with context.bzl

File: dart/build_rules/internal/common.bzl [hunk 4]
```diff
@@ -12,181 +12,44 @@
...
-"""Internal implemenation utility functions for Dart rules.
+"""Internal implementation utility functions for Dart rules.
 
 WARNING: NOT A PUBLIC API.
 
 This code is public only by virtue of the fact that Bazel does not yet support
-a mechanism for enforcing limitied visibility of Skylark rules. This code makes
-no gurantees of API stability and is intended solely for use by the Dart rules.
+a mechanism for enforcing limited visibility of Skylark rules. This code makes
+no guarantees of API stability and is intended solely for use by the Dart rules.
```
Comment (natebosch) [inline dart/build_rules/internal/common.bzl]: this makes it sound like we make guarantees of stability outside of internal... We've got the whole package marked with a big warning, maybe we can remove this comment?

File: dart/build_rules/internal/common.bzl [hunk 5]
```diff
@@ -12,181 +12,44 @@
...
 """
 
+# Re-exports for backwards compatibility.
```
Comment (natebosch) [inline dart/build_rules/internal/common.bzl]: How long do we plan on keeping the backwards compatible load location?

File: dart/build_rules/common/context.bzl (hunk 1 above)
Comment (jakemac53) [inline dart/build_rules/common/context.bzl]: Sure, done.

File: dart/build_rules/common/util.bzl (hunk 2 above)
Comment (jakemac53) [inline dart/build_rules/common/util.bzl]: done

File: dart/build_rules/common/util.bzl (hunk 3 above)
Comment (jakemac53) [inline dart/build_rules/common/util.bzl]: done

File: dart/build_rules/internal/common.bzl (hunk 4 above)
Comment (jakemac53) [inline dart/build_rules/internal/common.bzl]: agreed, done

File: dart/build_rules/internal/common.bzl (hunk 5 above)
Comment (jakemac53) [inline dart/build_rules/internal/common.bzl]: Not long I don't think, I just want to do that as a separate step given the fairly large structural changes already in this pr.

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
Note that I broke out the `label_to_dart_package_name` helper into its own file, as we have some additional logic internally so this needs to be overridden.

**Review by natebosch — APPROVED** [review]

**Pull request merged**